features/bootstrap/DatasetViewContext.php

```diff
@@ -217,6 +217,15 @@ public function iShouldSeeAButton($arg1)
         PHPUnit_Framework_Assert::assertTrue( $this->minkContext->getSession()->getPage()->hasLink($arg1) );
     }
 
+    /**
+     * @Then I manually hit return
+     */
+    public function iManuallyHitReturn()
+    {
+        $xpath = '//*[@id="pageNumber"]';
+        $this->minkContext->getSession()->getDriver()->getWebDriverSession()->element('xpath', $xpath)->postValue(['value' => ["\r\n"]]);
+    }
```

Comment:
At the moment, this function seems quite restrictive as it can only be used to simulate a Return key press in a text field with id = "pageNumber". The function could be made more generic to simulate any key press provided as a parameter on any field which can be located with an XPath string.

Maybe the php-doc annotation could be something like this: `@And I send the :arg1 field with keypress :arg2` and the function name `iSendTheFieldWithKeyPress($arg1, $arg2)`.

Reply:
I would keep the step syntax and the name of the implementing function specific to scenario as it is for now (although I'd drop the "manually") but factor out the logic into a more generic ``sendKeyPressFromField()`` helper function. 

```
/**
 * @When I hit return
 */
function IHitReturn() {
	$this->sendKeyPressFromField("\r\n","pageNumber")
}
```

the scenario reads better that way from user/business perspective, yet the main logic is generic:
```
When I fill in "pageNumber" with "2"
And I hit return
```

Later, on future work, if a similar need for pressing a key on a different field arises, we could always revisit this and come up with better abstraction that fit that future context and this one. But until then, it's still hypothetical (there are many ways to abstract and generize, doing it too early may get us the wrong one)

Reply:
@rija I followed your suggestion, and ran the behat test, but it failed and thrown the following error:
```
 And I hit return                                           # DatasetViewContext::iHitReturn()
      Fatal error: Call to undefined method DatasetViewContext::sendKeyPressFromField() (Behat\Testwork\Call\Exception\FatalThrowableError)
    │
    │  http://gigadb.dev/dataset/101001
    │
    └─ @AfterStep # GigadbWebsiteContext::debugStep()
```

Reply:
@kencho51 It looks like you need to create the `sendKeyPressFromField($keypress, $field_name)` function in your DatasetViewContext class.

Reply:
@pli888 `sendKeyPressFromField($key, $field_name)` has been created and passed the Behat test.